src/main/kotlin/com/jetbrains/notary/extensions/notarize.kt

```diff
@@ -116,48 +118,51 @@ private suspend fun NotaryClientV2.awaitSubmissionCompletion(
-                    logger.warn("Ignoring call failure to Notary API, will check status again in ${pollingConfiguration.retryDelayAfterFailure}:\n$e")
-
-                (pollingConfiguration.ignoreTimeoutExceptions && e.isTimeoutException()) ->
-                    logger.warn("Ignoring call timeout to Notary API, will check status again in ${pollingConfiguration.retryDelayAfterFailure}:\n$e")
-
-                else -> throw e
+    coroutineScope {
+        while (isActive) {
+            val response = try {
+                getSubmissionStatus(submissionId)
+            } catch (e: Exception) {
+                when {
+                    (pollingConfiguration.ignoreServerError && e is ServerResponseException) ->
+                        logger.warn("Ignoring call failure to Notary API, will check status again in ${pollingConfiguration.retryDelayAfterFailure}:\n$e")
+
+                    (pollingConfiguration.ignoreTimeoutExceptions && e.isTimeoutException()) ->
+                        logger.warn("Ignoring call timeout to Notary API, will check status again in ${pollingConfiguration.retryDelayAfterFailure}:\n$e")
+
+                    else -> throw e
+                }
+                delay(pollingConfiguration.retryDelayAfterFailure)
+                continue
             }
-            delay(pollingConfiguration.retryDelayAfterFailure)
-            continue
-        }
-        when (val status = response.data?.attributes?.status) {
-            SubmissionResponse.Status.ACCEPTED,
-            SubmissionResponse.Status.INVALID,
-            SubmissionResponse.Status.REJECTED -> return status
+            when (val status = response.data?.attributes?.status) {
+                SubmissionResponse.Status.ACCEPTED,
+                SubmissionResponse.Status.INVALID,
+                SubmissionResponse.Status.REJECTED -> return@coroutineScope status
 
-            SubmissionResponse.Status.IN_PROGRESS -> logger.info("Notarization still in progress, will check status again in ${pollingConfiguration.pollingPeriod}")
-            null -> logger.warn("Notarization status unknown, will check status again in ${pollingConfiguration.pollingPeriod}")
+                SubmissionResponse.Status.IN_PROGRESS -> logger.info("Notarization still in progress, will check status again in ${pollingConfiguration.pollingPeriod}")
+                null -> logger.warn("Notarization status unknown, will check status again in ${pollingConfiguration.pollingPeriod}")
+            }
+            delay(pollingConfiguration.pollingPeriod)
         }
-        delay(pollingConfiguration.pollingPeriod)
     }
+    error("Unable to complete notarization for submission $submissionId")
 }
 
 private fun Exception.isTimeoutException(): Boolean = this is HttpRequestTimeoutException
         || this is SocketTimeoutException
         || this is ConnectTimeoutException
 
-private fun sha256(path: Path): String {
+private suspend fun sha256(path: Path): String = coroutineScope {
```

Comment:
Could we use `yield()` inside the `while` (at the end) instead? `coroutineScope` + `isActive` is a bit verbose